File: Atributos de validación Data Annotations multiidioma/[VB]-Atributos de validación Data Annotations multiidioma/VB/LocalizeDataAnnotations/Infrastructure/CustomDisplayAttribute.vb
Imports System.ComponentModel

Public Class CustomDisplayAttribute
    Inherits DisplayNameAttribute

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(displayName As String)
        MyBase.New(displayName)
    End Sub

    Public Overrides ReadOnly Property DisplayName As String
        Get
            Dim name As String = TextosBBDD.Recuperar(MyBase.DisplayName)
            Return If(name, MyBase.DisplayName)
        End Get
    End Property

End Class

File: Atributos de validación Data Annotations multiidioma/[C#]-Atributos de validación Data Annotations multiidioma/C#/LocalizeDataAnnotations/Infrastructure/CustomDisplayAttribute.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Web;

namespace LocalizeDataAnnotations.Infrastructure
{

    public class CustomDisplayAttribute : DisplayNameAttribute
    {

        public CustomDisplayAttribute(): base() { }

        public CustomDisplayAttribute(string displayName): base(displayName){ }

        public override string DisplayName
        {
            get
            {
                string displayName = TextosBBDD.Recuperar(base.DisplayName);
                return displayName ?? base.DisplayName;
            }
        }
        

    }
}